WordPress/src/main/java/org/wordpress/android/ui/DeepLinkingIntentReceiverActivity.java

```diff
@@ -57,41 +71,136 @@ protected void onCreate(Bundle savedInstanceState) {
                     mInterceptType = InterceptType.VIEWPOST;
                     mBlogId = uri.getQueryParameter("blogId");
                     mPostId = uri.getQueryParameter("postId");
-                    break;
+
+                    // if user is logged in, show the post right away - otherwise show welcome activity
+                    // and then show the post once the user has logged in
+                    if (AccountHelper.isSignedInWordPressDotCom()) {
+                        showPost();
+                    } else {
+                        Intent intent = new Intent(this, SignInActivity.class);
+                        startActivityForResult(intent, INTENT_WELCOME);
+                    }
+                    return;
                 case "http":
                 case "https":
+                    mInterceptedUri = uri.toString();
+
                     List<String> segments = uri.getPathSegments();
 
                     // Handled URLs look like this: http[s]://wordpress.com/read/feeds/{feedId}/posts/{feedItemId}
                     //  with the first segment being 'read'.
-                    if (segments != null && segments.get(0).equals("read")) {
-                        if (segments.size() > 2) {
-                            mBlogId = segments.get(2);
-
-                            if (segments.get(1).equals("blogs")) {
-                                mInterceptType = InterceptType.READER_BLOG;
-                            } else if (segments.get(1).equals("feeds")) {
-                                mInterceptType = InterceptType.READER_FEED;
+                    if (segments != null) {
+                        if (segments.get(0).equals("read")) {
+                            if (segments.size() > 2) {
+                                mBlogId = segments.get(2);
+
+                                if (segments.get(1).equals("blogs")) {
+                                    mInterceptType = InterceptType.READER_BLOG;
+                                } else if (segments.get(1).equals("feeds")) {
+                                    mInterceptType = InterceptType.READER_FEED;
+                                }
                             }
-                        }
 
-                        if (segments.size() > 4 && segments.get(3).equals("posts")) {
-                            mPostId = segments.get(4);
+                            if (segments.size() > 4 && segments.get(3).equals("posts")) {
+                                mPostId = segments.get(4);
+                            }
+
+                            parseFragment(uri);
+
+                            showPost();
+                            return;
+                        } else if (segments.size() == 4) {
+                            mBlogId = uri.getHost();
```

Comment:
Using `mBlogId` and `mPostId` here is misleading because they're being set to the slugs rather than the actual IDs. I can see this causing confusion in the future.